benchmarks/scripts/expval_benchmark.py

```diff
@@ -1,99 +1,216 @@
+            ucc_compiled.save_density_matrix()
+            return ucc_compiled
+
+        case "pytket":
+            pytket_compiled = pytket.qasm.circuit_to_qasm_str(
+                pytket_compile(circuit)
+            )
+            pytket_compiled_qiskit = qasm2.loads(pytket_compiled)
+            pytket_compiled_qiskit.save_density_matrix()
+            return pytket_compiled_qiskit
+
+        case "qiskit":
+            qiskit_compiled = qiskit_compile(circuit)
+            qiskit_compiled.save_density_matrix()
+            return qiskit_compiled
+
+        case "cirq":
+            cirq_compiled_qasm = cirq.qasm(cirq_compile(circuit))
+            cirq_compiled_qiskit = qasm2.loads(cirq_compiled_qasm)
+            cirq_compiled_qiskit.save_density_matrix()
+            return cirq_compiled_qiskit
+
+        case _:
+            raise ValueError(f"Unknown compiler alias: {compiler_alias}")
 
-    ucc_compiled = ucc_compile(qasm, return_format="qiskit")
-    ucc_compiled.save_density_matrix()
 
-    cirq_compiled = cirq.qasm(cirq_compile(circuit_from_qasm(qasm)))
-    cirq_compiled_qiskit = qasm2.loads(cirq_compiled)
-    cirq_compiled_qiskit.save_density_matrix()
+def simulate_density_matrix(circuit: qiskit.QuantumCircuit) -> np.ndarray:
+    """Simulates the given quantum circuit using a density matrix simulator
+    with depolarizing noise.
+
+    Args:
+        circuit: The quantum circuit to simulate.
 
-    pytket_compiled = pytket.qasm.circuit_to_qasm_str(
-        pytket_compile(pytket.qasm.circuit_from_qasm_str(qasm_string))
+    Returns:
+        The resulting density matrix from the simulation.
+    """
+    depolarizing_noise = create_depolarizing_noise_model(
+        circuit, SINGLE_QUBIT_ERROR_RATE, TWO_QUBIT_ERROR_RATE
+    )
+    simulator = AerSimulator(
+        method="density_matrix", noise_model=depolarizing_noise
     )
-    pytket_compiled_qiskit = qasm2.loads(pytket_compiled)
-    pytket_compiled_qiskit.save_density_matrix()
+    return simulator.run(circuit).result().data()["density_matrix"]
 
-    return {
-        "uncompiled": uncompiled_circuit,
-        "ucc": ucc_compiled,
-        "cirq": cirq_compiled_qiskit,
-        "qiskit": qiskit_compiled,
-        "pytket": pytket_compiled_qiskit,
-    }
 
+def qiskit_gateset(circuit: qiskit.QuantumCircuit) -> set[str]:
+    everything = set(circuit.count_ops().keys())
+    return everything - {"save_density_matrix"}
 
-def simulate_density_matrix(circuit: qiskit.QuantumCircuit) -> np.ndarray:
-    depolarizing_noise = NoiseModel()
-    error = depolarizing_error(0.03, 1)
-    two_qubit_error = depolarizing_error(0.05, 2)
-    depolarizing_noise.add_all_qubit_quantum_error(error, ["u1", "u2", "u3"])
-    depolarizing_noise.add_all_qubit_quantum_error(two_qubit_error, ["cx"])
 
-    simulator = AerSimulator(
-        method="density_matrix", noise_model=depolarizing_noise
+def parse_arguments() -> tuple[str, str, str, bool]:
```

Comment:
Another style nit for the future -- we should consider `argparse` or `click` for managing command line parsing.